TWLight/resources/views.py

```diff
@@ -240,3 +245,81 @@ def get_context_data(self, **kwargs):
             context['partner_streams'] = False
 
         return context
+
+
+
+class PartnerSuggestionView(FormView):
+    model=Suggestion
+    template_name = 'resources/suggest.html'
+    form_class = SuggestionForm
+    
+    def get_queryset(self):
+            return Suggestion.objects.order_by('suggested_company_name')
+    
+    def form_valid(self, form):
+        # Right now, we only hide the suggestion form on the template.
+        # Adding an extra check to ensure the user is looged in
+        # so as to not break things.
+        try:
+            assert self.request.user.is_authenticated()
+            suggestion = Suggestion()
+            suggestion.suggested_company_name = form.cleaned_data['suggested_company_name']
+            suggestion.description = form.cleaned_data['description']
+            suggestion.company_url = form.cleaned_data['company_url']
+            suggestion.author = self.request.user
+            suggestion.save()
+            suggestion.upvoted_users.add(self.request.user)
+            messages.add_message(self.request, messages.SUCCESS,
+            # Translators: Shown to users when they successfully add a new partner suggestion.
+            _('Your suggestion has been added.'))
+            return HttpResponseRedirect(reverse('suggest'))
+        except AssertionError:
+            messages.add_message (request, messages.WARNING,
+                # Translators: This message is shown to users who attempt to post data to suggestion form without logging in
+                _('You must be logged in to do that.'))
+            raise PermissionDenied
```

Comment:
I know we ended up with this after discussing how the display logic should work, but now that I see it in action, I think we might be better served by either:
a) wrapping the whole page in the login redirect machinery like everything else, or
b) wrapping just the form post like that.

Reply:
@uyscuti-wiki I'm throwing a little bit of a wrench into some code that works exactly as we all discussed. If the thought of working on this any more makes you want to throw your hands up, let me know and I can bang out this piece.

Reply:
I'm unsure as to how to proceed without some additional help. Don't think the first option would work since we are interested in allowing unauthenticated users to browse the existing suggestions. As for the second option, is it a different method I'd have to override? Or is it something else? 
If you think it's too much trouble to explain, I'm okay with "banging out this piece" (not sure what that is either) :)

Reply:
Yeah you're right about option a not really being an option, which is why we landed where we are. For option b, you just need to make sure that the view that processes the form post has its own class-based url defined, then you can add the login_required decorator to that url. Here's an example from graphs:
https://github.com/WikipediaLibrary/TWLight/blob/c3834b5dd40c31e12db09672f240918cbfb0b5c9/TWLight/graphs/urls.py#L2
https://github.com/WikipediaLibrary/TWLight/blob/c3834b5dd40c31e12db09672f240918cbfb0b5c9/TWLight/graphs/urls.py#L43-L46

"banging it out" is an example of not-super helpful helpful idiom that I lapse into sometimes.
https://idioms.thefreedictionary.com/bang+out#Definition
Apologies.